src/concrete/ml/quantization/linear_op_glwe_backend.py

```diff
@@ -92,19 +190,25 @@ def _per_channel_weight_quantization(
             tuple: Quantized weights, scale, zero point and weight sum
         """
         weight_float = torch.from_numpy(weight).to(device)
-        q_min, q_max = self._get_quant_range(q_module)
+
+        # Get the signed integer range
+        q_min, q_max = self._get_quant_range(q_module, is_signed=True)
 
         w_min_vals, _ = weight_float.min(dim=0, keepdim=True)
         w_max_vals, _ = weight_float.max(dim=0, keepdim=True)
 
         weight_scale = (w_max_vals - w_min_vals) / (q_max - q_min)
+
         # Avoid division by zero.
         weight_scale = torch.where(
             (w_max_vals > w_min_vals), weight_scale, torch.ones_like(weight_scale)
         )
         weight_scale = weight_scale.squeeze(-1)  # shape: (out_dim,)
 
-        weight_zp = torch.round((q_min - w_min_vals) / weight_scale).to(torch.float32)
+        # Quantization
+        weight_zp = torch.round(q_min - w_min_vals / weight_scale).to(torch.float32)
```

Comment:
couldn't you use the TorchUniformQuantizer here? (maybe just TODO in a next PR, no need to fix now)